test/e2e_node/security_context_test.go

```diff
@@ -274,4 +275,51 @@ var _ = framework.KubeDescribe("Security Context", func() {
 		})
 	})
 
+	Context("When creating a container with runAsUser", func() {
+		makeUserPod := func(podName, image string, command []string, userid int64) *v1.Pod {
+			user := types.UnixUserID(userid)
+			return &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: podName,
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyNever,
+					Containers: []v1.Container{
+						{
+							Image:   image,
+							Name:    podName,
+							Command: command,
+							SecurityContext: &v1.SecurityContext{
+								RunAsUser: &user,
+							},
+						},
+					},
+				},
+			}
+		}
+		createAndWaitUserPod := func(podName string, userid int64) {
+			podClient.Create(makeUserPod(podName,
+				"gcr.io/google_containers/busybox:1.24",
+				[]string{"id", "-u"},
```

Comment:
rather than parsing the logs, just perform the test here. E.g.
`[]string{"test", "$(id -u)", "-eq", strconv.FormatInt(userid, 10)}`